docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,692 @@
+
+The `BOOLEAN` is a type which is available in PostgreSQL. In many other DBs, `TINYINTEGER` or `TINYINT` is commonly used
+instead (they are the same thing). The possible negative values are `false` and `0`, and the possible positive values 
+are `true`, and `1`.
+
+#### ENUM
+
+The `ENUM` type allows you to specify a small set of possible values for a column. In PostgreSQL, enums are created as
+a new datatype and then used in tables afterward. In other DB management systems, enums are usually just written as 
+you're declaring a column type.
+
+Here's an example of creating an enum type, which is shown used in a table later on:
+```sql
+CREATE TYPE personality AS ENUM ('nice', 'mean', 'neither');
+```
+
+### Column Modifiers
+
+Column modifiers are special modifiers you add when declaring a column in a table which give it extra properties. Here
+are a few of the common ones (some that you may need soon too!).
+
+#### PRIMARY KEY
+
+The `PRIMARY KEY` modifier declares a column or combination of columns as the ID for a given row. The database also uses
+a `PRIMARY KEY` to speed up some search functions that use those fields (see 
+[Indexes](https://www.postgresql.org/docs/9.1/indexes.html) if you're interested ). If used, all primary 
+keys must be unique on the given table. This column is also not nullable.
+
+#### FOREIGN KEY
+
+The `FOREIGN KEY` modifier is very similar to a `PRIMARY KEY` in that it defines a unique way to identify records on a 
+table. The main difference, though, is that a foreign key creates a link between a given table's columns and another
+table's `PRIMARY KEY`. We'll talk about this more in the next lesson.
+
+#### NOT NULL
+
+The `NOT NULL` modifier lets the database know that a column should never accept null values. You should try to use this
+where you can, since it can help with preventing nulls from being entered where you're not expecting them.
+
+#### AUTO INCREMENT
+
+The `AUTO INCREMENT` modifier lets the database that it should automatically enter this column as the default value
+(last value + 1) if the value isn't provided when entering data into the table. This is really useful for integer-typed
+`PRIMARY KEYS` in other databases, since it automatically creates a new unique ID for you.
+
+>Note: on PostgreSQL, this is actually 
+>the data type `SERIAL` instead, and the `AUTO INCREMENT` modifier isn't available.
+
+#### DEFAULT
+
+The `DEFAULT` modifier lets you specify a default value for this column if nothing is entered. For example, saying 
+`DEFAULT 0` will insert 0 if no value is provided for this column on insert. 
+
+### Creating a Database Table
+
+To create a database, all you have to do is run one of the following commands.
+
+```sql
+CREATE DATABASE your_database_name;
+
+-- Do it, but perform a check to make sure it doesn't exist first. If it does, it will throw an error.
+CREATE DATABASE IF NOT EXISTS your_database_name;
+```
+
+To delete a database, the command is very similar.
```

Comment:
This is small but make a quick note that deleting a database can be bad and to only do it when you are positive you want to delete it.

Reply:
ooh good idea